sdk/appconfiguration/app-configuration/src/secretReference.ts

```diff
@@ -31,7 +31,7 @@ export const SecretReferenceHelper = {
   toConfigurationSettingParam: (
     secretReference: ConfigurationSettingParam<SecretReferenceValue>
   ): ConfigurationSettingParam => {
-    if (!secretReference.value) {
+    if (!secretReference.value || typeof secretReference.value.secretId !== "string") {
```

Comment:
At some point we have to trust the type system :)

The only way you can get in here is if the type has been vetted (by a typeguard), right? If so, I think you can drop the extra type checks, unless your typeguard doesn't cover those.

Reply:
removed